docs/architecture/ADR-009-liquidity-mining.md

```diff
@@ -0,0 +1,175 @@
+# ADR 009: Liquidity Mining
+
+## Changelog
+
+- May 1 2022: Initial Draft (@toteki)
+
+## Status
+
+Proposed
+
+## Abstract
+
+Umee wishes to add support for liquidity mining incentives; i.e. additional rewards on top of the normal `x/leverage` lending APY for supplying base assets.
+
+For example, a user might "lock" for 14 days some of their `u/uatom` collateral held in the leverage module, earning an additional 12% APY of the collateral's value, received as `uumee` tokens.
+
+Locked tokens will be unavailable for `x/leverage` withdrawal until unbonded, but still able to be liquidated. There will be 3 locking tiers, differing in unbonding duration.
+
+Incentive programs will be created by governance proposals, get funded with tokens, then (from `StartDate` to `EndDate`) distribute those tokens to lenders of based on the lenders' locked value and lock tier. APY will vary as fixed reward amounts are divided amongst all participating lenders.
+
+## Context
+
+- TODO
+
+### Backwards Compatibility
+
+This will introduce a new module as well as new behavior in the `x/leverage` module. The changes will take place during the mainnet `calypso` upgrade.
+
+## Decision
+
+The general approach will be to create an `x/incentive` module with a small surface of interaction with `x/leverage`.
+
+The incentive module will support message types which allow users to lock and begin unbonding uTokens, similar to staking. 
+
+Locked funds must be collateral-enabled uTokens. Locked funds will not leave their original place in the leverage module account. Locking will prevent withdrawal and collateral-disabling (but not liquidation) of the locked uTokens until they are successfully unbonded.
+
+There will be three tiers of locking, differing in their unbonding durations, which may receive differing incentives. The tiers will be of fixed durations (`1 | 7 | 14 days`) and will exist independent of active incentive programs.
+
+The intended structure of what is described as a single incentive program is as follows:
+- A fixed sum of a single token denomination
+- To be distributed evenly over time between a start date and an end date
+- To all addresses which have locked a specified uToken denomination
+- Proportional to their total value locked but not currently unbonding
+- Then weighted by locking tier
+- As calculated at the moment of distribution, either continuously of epoched
+
+All parameters mentioned above (dates, amounts, and denominations) must be set using a governance proposal, which creates the incentive program unless impossible under our chosen implementation.
```

Comment:
This should be done using 0.46 new gov handlers. So, let's say that we will have a concrete message which can only be executed by gov.

Reply:
Added a note in the (new) governance section about this. Can discuss further if needed when you review that addition.